roles/kubernetes/node/defaults/main.yml

```diff
@@ -6,7 +6,7 @@ kubelet_address: "{{ ip | default(fallback_ip) }}{{ (',' + ip6) if enable_dual_s
 kubelet_bind_address: "{{ ip | default('0.0.0.0') }}"
 
 # resolv.conf to base dns config
-kube_resolv_conf: "/etc/resolv.conf"
+kube_resolv_conf: "{{ '/run/systemd/resolve/resolv.conf' if 'systemd-resolved' in active_dns_services else '/etc/resolv.conf' }}"
```

Comment:
I know that some resolv.conf modes (static if I remember correctly) check `/etc/resolv.conf` for files.
Using the systemd-resolved enablement as a check doesn't seem to be accurate.

`kube_resolv_conf` is by default still `/etc/resolv.conf`.
If systemd-resolved is enabled and `/etc/resolv.conf` is a soft link, change `kube_resolv_conf` to `/run/systemd/resolve/resolv.conf`.

What do you think?